### DISCUSSION TITLE
v6.0.0 Release Candidate Discussion

### DISCUSSION THREAD

**Discussion by krzywon** [SasView/sasview#3116] [General]
Please report when you you have downloaded and tested RC2 (https://github.com/SasView/sasview/releases/tag/v6.0.0-rc2).

### System Information
Windows 10, Windows 11, MacOS (Silicon), MacOS (Intel), Linux (distro and version), or Other

### Installer Tested
setupSasView-6.0.0rc2-Win64.exe, SasView-6.0.0rc2-MacOS_x86.dmg, SasView-6.0.0rc2-MacOS_arm.dmg, SasView-6.0.0rc2-Linux.tar.gz, or from Source Code

### New Features Tested that are Working
CorFunc, including results export, Preferences, Orientation Viewer, Plot Preferences (and plot suppression), Wedge Slicer, Center of Slicer not at (0,0) (box slicers only)

### Other Features Tested
Please provide a list of features tested, eg. Fitting, Constrained Fitting, Keissig Calculator, Read the documentation, etc.

### Issues found
Please provide a list of any issues found

**Comment by krzywon** [comment]
## System Information
Windows 10

## Installer Tested
setupSasView-6.0.0rc2-Win64.exe

## New Features Tested that are Working
All plotting preferences work as advertised.
Corfunc seems stable for all various options selected.
Documentation window opens expected windows.

## Other Features Tested
Fitting seems fine. PD is behaving with multiplicity models.

## Issues found
Using the GSC empirical model to generate a theory results in the following error. Fitting a data set to the model does not produce the same error. Other theory models work fine.
>  ERROR: Traceback (most recent call last): File "sas\qtgui\MainWindow\GuiManager.py", line 1360, in showPlot File "sas\qtgui\MainWindow\DataExplorer.py", line 1161, in displayData File "sas\qtgui\MainWindow\DataExplorer.py", line 1212, in plotData File "sas\qtgui\Plotting\Plotter.py", line 311, in plot File "matplotlib\axes\_base.py", line 3871, in set_ybound File "matplotlib\axes\_base.py", line 3973, in set_ylim File "matplotlib\axis.py", line 1236, in _set_lim File "matplotlib\axes\_base.py", line 3660, in _validate_converted_limits ValueError: Axis limits cannot be NaN or Inf

**Reply by krzywon** [reply]
After some more testing, this was an issue noted in the original PR (https://github.com/SasView/sasview/pull/2565). If the fitting theory limits are outside the original GSC calculation range, the dQ values are set to `nan`. This is documented, so this might not need to be a blocker. I'll see if there is a way to add Q range limits when loading the model, otherwise, this might have to get punted to a later release.

**Reply by krzywon** [reply]
A fix for this issue is currently in #3122.

**Comment by smk78** [comment]
### System Information
Windows 10

### Installer Tested
setupSasView-6.0.0rc2-Win64.exe

### New Features Tested that are Working
Corfunc.

### Other Features Tested
Data file i/o. All Tools _except_ Python shell. All Slicers. Copy/paste params. Save params (both). Report Results. Save/open project.

### Issues found
We still seem to have the plotting issue that if you change a plot title (not the plot window title) then go back and change some other setting - I was playing with a 2D plot and changed the colour scale limits - the plot title reverts to what it was originally. I don't think this is specific to 6 though, it happens in 5 too. Just saying. ***UPDATE*** Oh... actually you cannot change the title of a plot and then change, for example, the colour scale limits, or vice versa; you change one, it reverts the other! That's super annoying!

**Reply by krzywon** [reply]
The issue you see is present in v5, so not a 6.0.0-specific issue. Note - I can reproduce the title issue being reset when other things changes, but not the color map reset issue.

**Reply by smk78** [reply]
To reproduce: load some 2D data & plot it. Toggle the nav menu and click settings. Select the data and change the title. Click Apply & OK. Now right-click and select 2D Color Map. Change a Min or Max amplitude and click Apply. See what happens!

**Comment by smk78** [comment]
### System Information
Windows 10

### Installer Tested
setupSasView-6.0.0rc2-Win64.exe

### New Features Tested that are Working

### Other Features Tested
Simultaneous fitting

### Issues found
So this is a new one, I think. Went to test fitting by following the simultaneous fitting tutorial. All went well until I got to the part that said now change to the DREAM optimiser. So I went Fitting > Fit Algorithms and saw this:

![image](https://github.com/user-attachments/assets/daceef04-df20-475d-bc2f-6d415518f954)

If you look closely, it says the default optimiser is DREAM, and in fact won't let me Apply it. But the FitPage says the active optimiser is L-M.

Nothing I do will cause any FitPage to say DREAM and the speed of the fit & Result Panel is clearly not DREAM. In fact, it doesn't seem to want to change optimiser at all.

**Reply by smk78** [reply]
Ok, so I exited and restarted SasView and tried playing with changing the Fit Algorithm.

The issue is that when SasView starts it is not reading the default Fit Optimizer from preferences, it _always_ comes up as L-M regardless.

**Reply by krzywon** [reply]
This was originally reported as a part of alpha1 in https://github.com/SasView/sasview/issues/2889. The issue wasn't exactly fixed, but wasn't easily reproducible. I would also like to see this. The tutorial you were following that led to this is the one in https://github.com/SasView/tutorials/pull/15, correct?

**Reply by smk78** [reply]
It was, but is irrelevant to reproducing the bug. To reproduce:

Start SasView, Note the Active Fitting Optimizer in the FitPage is L-M. Go into Fitting > Fit Algorithms and change the Default Fit Algorithm to something other than L-M. Apply it. Notive the Active Fitting Optimizer changes. Exit SasView and restart. Notice that the Active Fitting Optimizer is reset to the L-M, even though the previous applied change of default is still true.

**Reply by krzywon** [reply]
Oh, that one is annoying and easily reproducible. I'll see if I can't fix it.

**Reply by krzywon** [reply]
Okay, that was an easy fix. I just pushed it to https://github.com/SasView/sasview/pull/3122. Give the installer a try when you are able.

**Reply by smk78** [reply]
Thanks @krzywon . Tested actions/runs/11292734190 on W10/x64. On starting SasView the FitPage is now assigned whatever the stored Default Fit Algorithm is. And once in the program, changing the Fit Algorithm changes what is used in the FitPage. In summary, the expected behaviour is now present.

**Comment by smk78** [comment]
Oh dear... I'm sorry to say that whilst testing the P(Q) @ S(Q) example in the basic fitting tutorial I have just encountered the bug reported in https://github.com/SasView/sasview/issues/2994. To be fair, this is using an example with a low salt concentration and as @krzywon pointed out in https://github.com/SasView/sasview/issues/3101#issuecomment-2334723676 issues had been previously noted, but my reading of https://github.com/SasView/sasview/issues/2994 is that the 'hanging' was supposed to be trapped?

**Reply by smk78** [reply]
But it's very hit-n-miss to reproduce.

**Reply by krzywon** [reply]
As noted in https://github.com/SasView/sasview/issues/2994#issuecomment-2299187573, various forms of this issue have been around since v5. I don't think we should hold up 6 for this.

**Comment by smk78** [comment]
Has the fact the GSC is opening behind the main GUI window been captured already? I don't see it in https://github.com/SasView/sasview/pull/3122.

**Reply by smk78** [reply]
Plot computed by GSC also appeared behind the GSC window!

**Reply by krzywon** [reply]
I've tried a number of things for the GSC opening behind the main window (`raise()`, `activateWindow()`, `setFocus()`, setting the parent to something else, and more) , but none of them seem to work. This might be one we have to punt on for now.

**Reply by smk78** [reply]
Could we put something in the message area once it has opened? 'General Scattering Calculator has opened but may be hidden'?

**Reply by krzywon** [reply]
Yes, in fact, I can! Check out the latest commit in the final fix branch.

**Reply by krzywon** [reply]
Side note - the plots always appear behind the GSC window. I don't think that is a bug.

**Reply by smk78** [reply]
> Yes, in fact, I can! Check out the latest commit in the final fix branch.

Tested actions/runs/11296492573 on W10/x64. Looks good!

**Comment by wpotrzebowski** [comment]
### System Information
Rocky LINUX

### Installer Tested
SasView-6.0.0-Linux.tar.gz

### Issues found
SasView starts but crashes after trying to load any data. No sensible error message only: aborted (core dumped).

**Reply by smk78** [reply]
@jamescrake-merani says by email:
>So interestingly, on KDE everything works fine. But on Gnome, it crashes immediately once I click on the load button (both for data, and projects).

>My suspicion is that, on KDE, it can open the dialog because KDE integrates closely with QT whereas Gnome doesn't so much; it mostly integrates with GTK. That shouldn't be causing this error though. I'm going to take a look at the logs on Gnome, and see if I can diagnose the problem.

**Reply by jamescrake-merani** [reply]
Here is the error I get. @wpotrzebowski is this the same as what you're getting?

```/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgioremote-volume-monitor.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgioremote-volume-monitor.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so

(sasview:1035158): Gtk-WARNING **: 15:44:16.500: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognised image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognised image file format (gdk-pixbuf-error-quark, 3)
fish: Job 1, './sasview' terminated by signal SIGABRT (Abort)
```

My theory is that, on KDE, since the native file chooser dialog is in QT, SasView has the necessary libraries, and icons to open it. But when on Gnome, the libraries necessary to open a GTK dialog aren't there, and so it just crashes.

**Comment by smk78** [comment]
### Things that need to happen for the Release

- [x] Ensure the contributor list is up-to-date
- [ ] Ensure the Release Notes are updated
- [x] Ensure the v6 tutorials are merged to tutorials\master and then copied to the v6 release branch

**Comment by wpotrzebowski** [comment]
### SciLifeLab logo is missing
Sorry I should have noticed it earlier but SciLifeLab logo should be added to About box. I will try to create a PR

**Comment by butlerpd** [comment]
Definitely need the SciLifeLab logo - they have been contributing a lot lately!

I hope others have had a chance to test things a little bit as I'll admit I've had zero time so fare.

**Comment by lucas-wilkins** [comment]
We know what the Linux issue is, as James fixed it on the flatpack. It doesn't seem possible to apply the same fix to the installer.

Re logos on the logo page. Remember that Qt sucks with bitmaps, so, make sure to resize it to the size the other non-svg ones are. If its an SVG then no problems.

**Comment by butlerpd** [comment]
I just noticed that there is a "particle editor" perspective in the release 6.0 branch? but I don't see it in the analysis list? Did we miss the link? Or are we shipping a bunch of unfinished stuff? If the latter I guess it shouldn't matter in terms of user experience (unless they are using the CLI I guess)? but seems bad form to be shipping experimental/unfinished stuff as part of a release? I've been deep underwater for a month so apologies if I've missed something that has already been discussed.

**Comment by jamescrake-merani** [comment]
I found a bug with the 2nd release candidate this morning: #3127

Its a fairly minor error but since the release is now being held back, I'll see if I can fix it before the release.

**Reply by jamescrake-merani** [reply]
Pull request submitted earlier today: #3128

**Comment by krzywon** [comment]
I am nearly there with #3126. When that and #3127 are fully resolved, I will finalize the release. Likely Monday at this point.
